models/upload_sessions.go

```diff
@@ -312,29 +313,56 @@ func (u *UploadSession) GetTreasureIndexes() ([]int, error) {
 
 func (u *UploadSession) BulkMarkDataMapsAsUnassigned() error {
 	var err error
+	var ids []string
+	var dm []DataMap
+
+	err = DB.RawQuery("SELECT id FROM data_maps WHERE genesis_hash = ? AND status = ? AND message != ? AND msg_status = ?",
+		u.GenesisHash,
+		Pending,
+		DataMap{}.Message,
+		MsgStatusUnmigrated).All(&dm)
+	oyster_utils.LogIfError(err, nil)
+
+	for _, v := range dm {
+		ids = append(ids, v.ID.String())
+	}
+	err = nil
+
+	idsString := strings.Join(ids, ", ")
+
 	for i := 0; i < oyster_utils.MAX_NUMBER_OF_SQL_RETRY; i++ {
+		if len(idsString) == 0 {
+			break
+		}
 		err = DB.RawQuery("UPDATE data_maps SET status = ? "+
-			"WHERE genesis_hash = ? AND status = ? AND message != ? AND msg_status = ?",
+			"WHERE id IN(?)",
 			Unassigned,
-			u.GenesisHash,
-			Pending,
-			DataMap{}.Message,
-			MsgStatusUnmigrated).All(&[]DataMap{})
+			idsString).All(&[]DataMap{})
```

Comment:
I have had terrible luck being able to use IN with Pop.  https://github.com/gobuffalo/pop#using-in-queries   

I was talking with someone in the buffalo slack and they said when you pass in something for IN, it has to be an interface.  `someInterface := make([]interface{}, len(whatever))`

And then in the query in Pop you'd use it like this `someInterface...`  

Not sure if this is helpful.  I never got it to work.

Reply:
I am going to give this up.
Since the IN() could be very big. And SQL will complain about again. I think if we lock the table, I think it will be fine

Reply:
And it is not easy to break the IN() into batches and add more complexity. Just trust the SQL db to do the work